runtime/src/main/java/org/corfudb/util/FileWatcher.java

```diff
@@ -110,6 +124,18 @@ private void poll() {
         }
     }
 
+    private String getFileHash() {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
```

Comment:
This is problematic, as far as i can say. MessageDigest is not thread safe. Please use apache utils instead

Reply:
UPD: nevermind, you create a new instance of MessageDigest each time, we should be ok then

Reply:
`DigestUtils` sounds like a good alternative. I think MessageDigest will be just fine given that the logic is guaranteed to run as a singleton on FileWatcher thread.